client/jetpack-connect/controller.js

```diff
@@ -381,3 +382,15 @@ export function redirectToLoginIfLoggedOut( context, next ) {
 
 	next();
 }
+
+export function redirectToSiteLessCheckoutIfLoggedOut( context, next ) {
+	const { type } = context.params;
+	const isLoggedIn = isUserLoggedIn( context.store.getState() );
+
+	if ( config.isEnabled( 'jetpack/siteless-checkout' ) && ! isLoggedIn ) {
```

Comment:
I'm not sure we should filter out logged-in users since the new checkout will have to support them. 

For example, a user that already has a WP.com will have to log in before checking out (the flow for they will be `checkout` -> `login` -> `checkout`). So, the checkout route should be available for logged-in users as well.

It could also happen that a user has a WP.com account and now wants to purchase a Jetpack for a new self-hosted site. We still want to let the user complete a siteless purchase (the request to connect their site will come after purchasing).

Please let me know if this makes sense to you. We're still figuring out things, so don't hesitate to raise your hand if a flow looks broken to you.

Reply:
That all makes sense. In my mind there was a big divide between logged in and logged out since that preserved more of the original connect flow, but now it is clear to me that we are _replacing_ more than sitting alongside the old connect flow.

With that in mind I think the correct approach is to completely change the `page` function I modified ( always redirecting ), instead of just padding it with some middleware.